idseq_pipeline/commands/common.py

```diff
@@ -623,3 +609,16 @@ def upload_version_tracker(source_file, output_name, reference_version_number, o
 
 def env_set_if_blank(key, value):
     os.environ[key] = os.environ.get(key, value)
+
+def validate_taxid_lineage(taxid_lineage, hit_taxid, hit_level):
+    # Take the taxon lineage and mark meaningless calls with fake taxids.
+    # For each hit, the fake taxids below the meaningful hit level should depend
+    # only on: (a) the taxonomic level (b) the taxid of the meaningful hit
+    cleaned_taxid_lineage = []
+    for i in range(len(taxid_lineage)):
+        taxid = taxid_lineage[i]
+        tax_level = i+1
+        if tax_level < int(hit_level):
```

Comment:
Here are some minor stylistic suggestions, and you are totally welcome to use them or ignore them;   my feeling is that suggestion 3 may be the more useful one.

1) When expecting an integer value to be passed as a string, you might name the argument `hit_taxid_str` or `hit_level_str` instead of `hit_taxid` and `hit_level`.

2.   Pull out of the loop anything that gets repeated on each iteration.
```
hit_level = int(hit_level_str)
hit_taxid = int(hit_taxid_str)
```

3. Combine the benefits of 1 and 2 above by making it the caller's responsibility to convert hit_taxid and hit_level to integers before passing those arguments into this function.   The arguments can then continue to be called `hit_taxid` and `hit_level`.

In Python3 you could also annotate the function to indicate that those arguments should be integers.   Such annotations in general help correctness and sometimes can help performance.

4.  For a pure function without side effects (that is, a function that does not use or modify any global state, and whose sole purpose is to return a result determined solely from its argument values), you could name it with a noun that describes the value it returns, just like you would a mathematical function;  whereas functions with side effects can be named with a verb.   So this function could be called `cleaned_taxid_lineage`.